### PR TITLE
fix: respect `https.cacert` option

### PR DESCRIPTION
**Pull request by snitin315**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No

### Motivation / Use-Case

Fix https://github.com/webpack/webpack-dev-server/issues/3814

### Breaking Changes
None

### Additional Info
https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options

### CODE REVIEW

**Comment by snitin315** [comment]
Let's add `ca` in a separate PR as it is a feature.  Or should we add it to this PR?

**Comment by alexander-akait** [comment]
I will finish PR, here some more work

File: lib/Server.js [hunk 1]
```diff
@@ -654,6 +655,18 @@ class Server {
         this.logger.info(`SSL certificate: ${certificatePath}`);
       }
 
+      if (options.https.cacert) {
+        if (options.https.ca) {
+          this.logger.warn(
+            "Do not specify 'https.ca' and 'https.cacert' options, the 'https.ca' option will be used"
```
Comment (snitin315) [inline lib/Server.js:661]:
```suggestion
            "Do not specify 'https.ca' and 'https.cacert' options together, the 'https.ca' option will be used."
```

**Comment by alexander-akait** [comment]
@snitin315 Need documented after merge:
- added the `http.ca` option (for cli too) (should be used instead `cacert`, because we remove it in the next major release)
- added the `https.crl` (for cli too) https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
- `https.ca`/`https.cacert`/ `https.cert`/`https.crl`/`https.key`/`https.pfx` options are now accept Arrays of `Buffer`/`string`/Path to file
- `https` option can now accept custom options, now you can use:
```
https: {
  // Allow to set additional TSL options https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
  minVersion: "TLSv1.1",
  ca: path.join(httpsCertificateDirectory, "ca.pem")),
  pfx: path.join(httpsCertificateDirectory, "server.pfx"),
  key: path.join(httpsCertificateDirectory, "server.key"),
  cert: path.join(httpsCertificateDirectory, "server.crt"),
  passphrase: "webpack-dev-server",
},
```

**Comment by snitin315** [comment]
@alexander-akait I will take care of documentation 👍🏻

**Pull request merged**